src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -752,6 +756,8 @@ private static void OnAssemblyLoad(RuntimeAssembly assembly)
 
         [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
             Justification = "Satellite assemblies have no code in them and loading is not a problem")]
+        [UnconditionalSuppressMessage("Single file", "IL3000: Avoid accessing Assembly file path when publishing as a single file",
+            Justification = "This call is fine because native call runs before this and checks BindSatelliteResourceFromBundle")]
```

Comment:
So the Justification is that this should never get called for single-file bundled assemblies, right?

Reply:
In that case the method should still handle null from `Assembly.Location` in some way - at the very least an assert, but maybe just return null if it's null. Currently it will NullRef which is pretty bad.

Reply:
Returning null will be just hiding a problem, I will put an assert instead

Reply:
Actually the code is already handling null
```C#
string? parentDirectory = Path.GetDirectoryName(parentAssembly.Location);
if (parentDirectory == null)
    return null;
```
GetDirectoryName checks for null and string.Empty and in that case returns null